src/utils/validators.js

```diff
@@ -0,0 +1,325 @@
+    throw new TypeError(_composeTypeErrorMessage({
+      value,
+      expectedType: 'array',
+      fieldName
+    }))
+  }
+
+  if (value.length < minLength) {
+    throw new Error(_composeErrorMessage({
+      value,
+      statement: `must have minimum length of ${minLength}`,
+      fieldName
+    }))
+  }
+}
+
+/**
+ * Validate value to be a string.
+ *
+ * @param {any} value The value to validate.
+ * @param {string} [fieldName] The name of the field, containing value.
+ * @param {number} [minLength] The minimum string length.
+ * @param {number} [maxLength] The maximum string length.
+ *
+ * @throws {TypeError} Value should be a valid string.
+ */
+export function validateString ({ value, fieldName = '', minLength, maxLength }) {
+  if (!BaseOperation.isValidString(value, minLength, maxLength)) {
+    throw new TypeError(_composeErrorMessage({
+      value,
+      statement: 'must be a valid string',
+      params: { minLength, maxLength },
+      fieldName
+    }))
+  }
+}
+
+/** Keys validators */
+
+/**
+ * Validate value to be a public key.
+ *
+ * @param {any} value The value to validate.
+ * @param {string} [fieldName] The name of the field, containing value.
+ *
+ * @throws {TypeError} Value should be a valid public key.
+ */
+export function validatePublicKey ({ value, fieldName = '' }) {
+  if (!Keypair.isValidPublicKey(value)) {
+    throw new TypeError(_composeTypeErrorMessage({
+      value,
+      expectedType: 'public key',
+      fieldName
+    }))
+  }
+}
+
+/**
+ * Validate value to be a secret key.
+ *
+ * @param {any} value The value to validate.
+ * @param {string} [fieldName] The name of the field, containing value.
+ *
+ * @throws {TypeError} Value should be a valid secret key.
+ */
+export function validateSecretKey ({ value, fieldName = '' }) {
+  if (!Keypair.isValidSecretKey(value)) {
+    throw new TypeError(_composeTypeErrorMessage({
+      value,
+      expectedType: 'secret key',
+      fieldName
+    }))
+  }
+}
+
+/**
+ * Validate value to be a balance key.
+ *
+ * @param {any} value The value to validate.
+ * @param {string} [fieldName] The name of the field, containing value.
+ *
+ * @throws {TypeError} Value should be a valid balance key.
+ */
+export function validateBalanceKey ({ value, fieldName = '' }) {
+  if (!Keypair.isValidBalanceKey(value)) {
+    throw new TypeError(_composeTypeErrorMessage({
+      value,
+      expectedType: 'balance key',
+      fieldName
+    }))
+  }
+}
+
+/** Fields validators */
+
+/**
+ * Validate value to be a int64 amount.
+ *
+ * @param {any} value The value to validate.
+ * @param {string} [fieldName] The name of the field, containing value.
+ * @param {boolean} [allowZero] Is zero amount allowed.
+ * @param {number} [min] The minimum amount.
+ * @param {number} [max] The maximum amount.
+ *
+ * @throws {TypeError} Value should be a valid int64 amount.
+ */
+export function validateAmount ({
+  value,
+  fieldName = '',
+  allowZero = false,
+  max,
+  min
+}) {
```

Comment:
same here about missing min/max properties

Reply:
`min` & `max` params are optional. If some of them are not passed, the validator ignores them. Examples of use can be found in specs.